receiver/haproxyreceiver/scraper.go

```diff
@@ -28,30 +32,60 @@ var (
 )
 
 type scraper struct {
-	endpoint string
-	logger   *zap.Logger
-	mb       *metadata.MetricsBuilder
+	cfg               *Config
+	httpClient        *http.Client
+	logger            *zap.Logger
+	mb                *metadata.MetricsBuilder
+	telemetrySettings component.TelemetrySettings
 }
 
 func (s *scraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
-	var d net.Dialer
-	c, err := d.DialContext(ctx, "unix", s.endpoint)
-	if err != nil {
-		return pmetric.NewMetrics(), err
-	}
-	defer func(c net.Conn) {
-		_ = c.Close()
-	}(c)
-	records, err := s.readStats(c)
-	if err != nil {
-		return pmetric.NewMetrics(), err
+
+	var records []map[string]string
+	if u, notURLerr := url.Parse(s.cfg.Endpoint); notURLerr == nil && (u.Scheme == "http" || u.Scheme == "https") {
```

Comment:
If you can, I suggest doing !strings.HasPrefix(u.Scheme, "http")` instead ?

Reply:
```suggestion
	if u, notURLerr := url.Parse(s.cfg.Endpoint); notURLerr == nil && !strings.HasPrefix(u.Scheme, "http") {
```